includes/assets/stories-widget.js

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+(function ($) {
+  /**
+   * Store field states in local variable.
+   *
+   * @type {Object}
+   */
+  var fieldState = window.webStoriesData || {};
+
+  /**
+   * Closure for making inputs reactive
+   * on view type selection change.
+   *
+   * @param {Object}  target Event object.
+   * @param {boolean} reset Flag to reset fields value.
+   */
+  var react = function (target, reset = false) {
+    var currentView = target.value;
+    var widget = target.closest('.widget');
+    var state = fieldState.fields[currentView];
+
+    for (var [key, value] of Object.entries(state)) {
+      var field = widget.querySelector('.' + key + '.stories-widget-field');
+      var fieldWrapper = widget.querySelector('.' + key + '_wrapper');
+      if (field && fieldWrapper && 'checkbox' === field.getAttribute('type')) {
+        if (reset) {
+          field.checked = false;
+        }
+        /**
+         * If this is readonly field.
+         * Assign the value automatically and hide it afterward.
+         */
+        if (value.readonly) {
+          field.checked = value.show;
+        }
+        fieldWrapper.style.display = value.readonly ? 'none' : 'block';
+      }
+    }
+  };
+
+  /**
+   * Returns the HTML Collection of view type selectors.
+   *
+   * @return {HTMLCollection} Array of elements.
+   */
+  var viewSelectors = function () {
+    return document.getElementsByClassName('view-type stories-widget-field');
```

Comment:
See comment here https://github.com/rtCamp/web-stories-wp/pull/189/files#r551891257

Reply:
Used ES6 everywhere; except for for jQuery event listeners.